core/src/main/java/io/kestra/core/tasks/flows/ForEachItem.java

```diff
@@ -239,135 +241,300 @@ public class ForEachItem extends Task implements ExecutableTask<ForEachItem.Outp
     @PluginProperty
     private final Boolean inheritLabels = false;
 
+    @Valid
+    private List<Task> errors;
+
+    @Override
+    public GraphCluster tasksTree(Execution execution, TaskRun taskRun, List<String> parentValues) throws IllegalVariableEvaluationException {
+        GraphCluster subGraph = new GraphCluster(this, taskRun, parentValues, RelationType.SEQUENTIAL);
+
+        GraphUtils.sequential(
+            subGraph,
+            this.getTasks(),
+            this.errors,
+            taskRun,
+            execution
+        );
+
+        return subGraph;
+    }
+
+    @Override
+    public List<Task> allChildTasks() {
+        return Stream
+            .concat(
+                this.getTasks() != null ? this.getTasks().stream() : Stream.empty(),
+                this.errors != null ? this.errors.stream() : Stream.empty()
+            )
+            .collect(Collectors.toList());
+    }
+
     @Override
-    public List<SubflowExecution<?>> createSubflowExecutions(
-        RunContext runContext,
-        FlowExecutorInterface flowExecutorInterface,
-        Flow currentFlow,
-        Execution currentExecution,
-        TaskRun currentTaskRun
-    ) throws InternalException {
-        var renderedUri = runContext.render(this.items);
-        if (!renderedUri.startsWith("kestra://")) {
-            var errorMessage = "Unable to split the items from " + renderedUri + ", this is not an internal storage URI!";
-            runContext.logger().error(errorMessage);
-            throw new IllegalArgumentException(errorMessage);
+    public List<ResolvedTask> childTasks(RunContext runContext, TaskRun parentTaskRun) throws IllegalVariableEvaluationException {
+        return FlowableUtils.resolveTasks(this.getTasks(), parentTaskRun);
+    }
+
+    @Override
+    public List<NextTaskRun> resolveNexts(RunContext runContext, Execution execution, TaskRun parentTaskRun) throws IllegalVariableEvaluationException {
+        return FlowableUtils.resolveSequentialNexts(
+            execution,
+            this.childTasks(runContext, parentTaskRun),
+            FlowableUtils.resolveTasks(this.errors, parentTaskRun),
+            parentTaskRun
+        );
+    }
+
+    public List<Task> getTasks() {
+        return List.of(
+            new ForEachItemSplit(this.getId(), this.items, this.batch),
+            new ForEachItemExecutable(this.getId(), this.inputs, this.inheritLabels, this.labels, this.wait, this.transmitFailed,
+                new ExecutableTask.SubflowId(this.namespace, this.flowId, Optional.ofNullable(this.revision))
+            ),
+            new ForEachItemMergeOutputs(this.getId())
+        );
+    }
+
+    @SuppressWarnings("unused")
+    public void setTasks(List<Task> tasks) {
```

Comment:
Seems to be weird

Reply:
but do we need to have a getTasks public ? I don't think so, this seems to be the main issue IMO?

Reply:
It is needed for the UI. If the list of tasks is not public the UI will not be aware of the subtasks of the flowable tasks so the special ForEachItem progress bar will not be displayed.
We may revisit the hack in the UI to avoid that.